src/llm/legacy/servable.cpp

```diff
@@ -0,0 +1,158 @@
+    if (legacyExecutionContext->payload.client->isDisconnected()) {
+        return absl::CancelledError();
+    }
+    legacyExecutionContext->finished.wait();
+    // std::lock_guard<std::mutex> guard(pipelineMutex);
+    // legacyExecutionContext->results = properties->pipeline->generate(legacyExecutionContext->inputIds, legacyExecutionContext->apiHandler->createGenerationConfig());
+    return absl::OkStatus();
+}
+
+absl::Status LegacyServable::parseRequest(std::shared_ptr<GenAiServableExecutionContext>& executionContext) {
+    executionContext->apiHandler = std::make_shared<OpenAIChatCompletionsHandler>(*executionContext->payload.parsedJson,
+        executionContext->endpoint,
+        std::chrono::system_clock::now(),
+        getProperties()->tokenizer);
+
+    auto status = executionContext->apiHandler->parseRequest(getProperties()->maxTokensLimit, getProperties()->bestOfLimit, getProperties()->isSpeculativePipeline);
+    if (!status.ok()) {
+        SPDLOG_LOGGER_ERROR(llm_calculator_logger, "Failed to parse request: {}", status.message());
+        return status;
+    }
+
+    if (executionContext->apiHandler->isStream()) {
+        executionContext->lastStreamerCallbackOutput = "";  // initialize with empty string
+        auto callback = [& lastStreamerCallbackOutput = executionContext->lastStreamerCallbackOutput](std::string text) {
+            SPDLOG_LOGGER_TRACE(llm_calculator_logger, "Streamer callback executed with text: [{}]", text);
+            lastStreamerCallbackOutput += text;
+            return ov::genai::StreamingStatus::RUNNING;
+        };
+
+        executionContext->textStreamer = std::make_shared<ov::genai::TextStreamer>(getProperties()->tokenizer, callback);
+    }
+    return absl::OkStatus();
+}
+
+absl::Status LegacyServable::prepareCompleteResponse(std::shared_ptr<GenAiServableExecutionContext>& executionContext) {
+    auto legacyExecutionContext = std::static_pointer_cast<LegacyServableExecutionContext>(executionContext);
+    if (legacyExecutionContext->payload.client->isDisconnected()) {
+        return absl::CancelledError();
+    }
+    executionContext->response = executionContext->apiHandler->serializeUnaryResponse(legacyExecutionContext->results);
+    SPDLOG_LOGGER_DEBUG(llm_calculator_logger, "Complete unary response: {}", executionContext->response);
+    return absl::OkStatus();
+}
+
+absl::Status LegacyServable::preparePartialResponse(std::shared_ptr<GenAiServableExecutionContext>& executionContext) {
+    auto legacyExecutionContext = std::static_pointer_cast<LegacyServableExecutionContext>(executionContext);
+    if (legacyExecutionContext->payload.client->isDisconnected()) {
+        return absl::CancelledError();
+    }
+    std::stringstream ss;
+    ss << executionContext->lastStreamerCallbackOutput;
+    std::string lastTextChunk = ss.str();
+    executionContext->lastStreamerCallbackOutput = "";
```

Comment:
executionContext->lastStreamerCallbackOutput is modified in another thread so I suppose we need some access protection